query/query/query.py

```diff
@@ -209,6 +202,7 @@ async def on_startup(app):
     app['jhc'] = jhc
 
     app['users'] = set()
+    app['sockets'] = set()
```

Comment:
what is this for?

Reply:
This tracks all of the currently-open websockets, with the idea that we'd close them down nicely if the server shuts down (see https://docs.aiohttp.org/en/stable/web_advanced.html#graceful-shutdown). I'd forgotten to implement the on_shutdown hook, though.